fea-rs/src/compile/tables/var_store.rs

```diff
@@ -0,0 +1,312 @@
+//! Building the ItemVariationStore
+//!
+//! this will probably get moved to write-fonts at some point?
+
+use std::collections::HashMap;
+
+use indexmap::IndexMap;
+use write_fonts::tables::variations::{
+    ItemVariationData, ItemVariationStore, VariationRegion, VariationRegionList,
+};
+
+///
+/// this probably ends up living in write-fonts
+#[derive(Clone, Debug, Default)]
+pub(crate) struct VariationStoreBuilder {
+    // region -> index map
+    all_regions: HashMap<VariationRegion, usize>,
+    // we use an index map so that we have a determenistic ordering,
+    // which lets us write better tests
+    delta_sets: IndexMap<RegionSet, IndexMap<DeltaSet, DeltaKey>>,
+    next_id: DeltaKey,
+}
+
+#[derive(Clone, Debug, Default)]
+pub(crate) struct VariationStoreKeyLookup {
+    map: HashMap<DeltaKey, DeltaKey>,
+}
```

Comment:
Neither typename nor comment nor the field "map" are informative as to what this is for. wdyt about a tuple-struct, an alternate name, and ideally two different types of key, e.g.

```rust
/// Maps temporary (builder) to final keys for a variation store
pub(crate) struct Remapping(HashMap<PreliminaryKey, DeltaKey>);

// hooray, PreliminaryKey (name subject to bikeshed) won't compile for use against a final variation store!
```